tracking/src/main/java/org/motechproject/nms/tracking/service/CacheService.java

```diff
@@ -0,0 +1,5 @@
+package org.motechproject.nms.tracking.service;
+
+public interface CacheService {
```

Comment:
I imagine we'll find one more place in the tracking module to cache, and
we'd then be adding one extra method to the CacheService class.
And I also imagine we'll do the same (create a CacheService class) in
kilkari and possibly other modules.

In the particular case of getEntityByClassName, you're right, it may very
well be added to the platform instead of being cached in this module.
But I want to proceed with perf.

Would it make sense to do a one pass after we're done optimizing and move
to platform what needs to be moved?

On Wed, Sep 16, 2015 at 11:58 AM, Rob LaRubbio <notifications@github.com>
wrote:

> In
> tracking/src/main/java/org/motechproject/nms/tracking/service/CacheService.java
> <https://github.com/motech-implementations/mim/pull/590#discussion_r39671014>
> :
>
> > @@ -0,0 +1,5 @@
> > +package org.motechproject.nms.tracking.service;
> > +
> > +public interface CacheService {
>
> ok, but this still feels like the wrong solution. I think the correct fix
> would be to add cacheing to the platform on the
> entityService.getEntityByClassName method. I don't see us doing that.
>
> Is there an existing service in the tracking module that we could add that
> method to that would make sense?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/motech-implementations/mim/pull/590/files#r39671014>.
>



-- 
Frank Huster
Mobile Health Innovations
Grameen Foundation
+1.206.325.6690 x201